### PR TITLE
[Bug fix] fix subscription not found issue and remove duplicate prompt

### PR DESCRIPTION
**Pull request by andxu**
cleanup annoying logs too

### CODE REVIEW

File: azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java [hunk 1]
```diff
@@ -364,9 +366,11 @@ protected Account getAzureAccount() throws MavenDecryptException, AzureExecution
 
     private Account login(@Nonnull com.microsoft.azure.toolkit.lib.auth.model.AuthConfiguration auth) {
         promptAzureEnvironment(auth.getEnvironment());
+        MavenAuthUtils.disableIdentityLogs();
         accountLogin(auth);
         final Account account = com.microsoft.azure.toolkit.lib.Azure.az(AzureAccount.class).account();
-        if (account.getAuthType() == AuthType.OAUTH2 || account.getAuthType() == AuthType.DEVICE_CODE) {
+        final boolean isInteractiveLogin = account.getAuthType() == AuthType.OAUTH2 || account.getAuthType() == AuthType.DEVICE_CODE;
+        if (isInteractiveLogin) {
```
Comment (Flanker32) [inline azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java:373]: Seems we could remove `if (isInteractiveLogin) {` here as Device code must be interactive

**Review by Flanker32 — DISMISSED** [review]

File: azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java [hunk 2]
```diff
@@ -459,8 +463,22 @@ private static Account doServicePrincipalLogin(com.microsoft.azure.toolkit.lib.a
         });
     }
 
-    private static void printCredentialDescription(Account account) {
-        System.out.println(account.toString());
+    private static void printCredentialDescription(Account account, boolean skipType) {
+        if (skipType) {
+            if (CollectionUtils.isNotEmpty(account.getSubscriptions())) {
```
Comment (Flanker32) [inline azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java:468]: Duplicate codes with `Account.toString()`, we may add `getSubscriptonDescription` and `getAuthTypeDescription()` later in `Account`

Comment (andxu) [reply azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java:466]: will refactor all the usage to account in plugin-lib

File: azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/springcloud/ConfigMojo.java [hunk 3]
```diff
@@ -343,6 +344,7 @@ private String promptSubscription() throws IOException, InvalidConfigurationExce
         this.wrapper.putCommonVariable("subscriptions", subscriptions);
         final Subscription select = this.wrapper.handleSelectOne("select-subscriptions", subscriptions, null,
             t -> String.format("%s (%s)", t.getName(), t.getId()));
+        com.microsoft.azure.toolkit.lib.Azure.az(AzureAccount.class).account().selectSubscription(Collections.singletonList(select.getId()));
```
Comment (wangmingliang-ms) [inline azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/springcloud/ConfigMojo.java:347]: no need to use qualified name.

File: azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java (hunk 2 above)
Comment (wangmingliang-ms) [reply azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java:466]: why not fix to use account.toString() while you at it.

File: azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/springcloud/ConfigMojo.java (hunk 3 above)
Comment (andxu) [reply azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/springcloud/ConfigMojo.java:347]: a lot of such usage, we need to handle them together

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/AbstractWebAppMojo.java [hunk 4]
```diff
@@ -416,6 +417,7 @@ protected AzureAppService getOrCreateAzureAppServiceClient() throws AzureExecuti
                 final List<Subscription> subscriptions = account.getSubscriptions();
                 final String targetSubscriptionId = getTargetSubscriptionId(getSubscriptionId(), subscriptions, account.getSelectedSubscriptions());
                 checkSubscription(subscriptions, targetSubscriptionId);
+                com.microsoft.azure.toolkit.lib.Azure.az(AzureAccount.class).account().selectSubscription(Collections.singletonList(targetSubscriptionId));
```
[the remark this answers is not in the document]
Comment (andxu) [reply azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/AbstractWebAppMojo.java:420]: a lot of such usage, we need to handle them together

**Review by wangmingliang-ms — DISMISSED** [review]

File: azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java (hunk 2 above)
Comment (andxu) [reply azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java:463]: because two auth type will be print

**Review by wangmingliang-ms — APPROVED** [review]

**Pull request merged**